File: java/space-age/src/main/java/SpaceAge.java
class SpaceAge {
    private enum Planet {
        MERCURY(0.2408467),
        VENUS(0.61519726),
        EARTH(1.0),
        MARS(1.8808158),
        JUPITER(11.862615),
        SATURN(29.447498),
        URANUS(84.016846),
        NEPTUNE(164.7913);

        private final double orbitalFactor;

        private Planet(double orbitalFactor) {
          this.orbitalFactor = orbitalFactor;
        }
    }

    private static final double EARTH_ORBITAL_PERIOD = 31557600.0;
    private final double seconds;

    SpaceAge(double seconds) {
      this.seconds = seconds;
    }

    double getSeconds() {
      return this.seconds;
    }

    double onEarth() {
      return on(Planet.EARTH);
    }

    double onMercury() {
      return on(Planet.MERCURY);
    }

    double onVenus() {
      return on(Planet.VENUS);
    }

    double onMars() {
      return on(Planet.MARS);
    }

    double onJupiter() {
      return on(Planet.JUPITER);
    }

    double onSaturn() {
      return on(Planet.SATURN);
    }

    double onUranus() {
      return on(Planet.URANUS);
    }

    double onNeptune() {
      return on(Planet.NEPTUNE);
    }

    private double on(Planet planet) {
      return this.seconds / EARTH_ORBITAL_PERIOD / planet.orbitalFactor;
    }
}

File: python/space-age/space_age.py
from typing import Dict
from enum import Enum, auto


class SpaceAge:
    class Planet(Enum):
        MERCURY = auto()
        VENUS = auto()
        EARTH = auto()
        MARS = auto()
        JUPITER = auto()
        SATURN = auto()
        URANUS = auto()
        NEPTUNE = auto()

    __EARTH_ORBITAL_PERIOD: float = 31557600.0
    __ORBITAL_FACTORS: Dict[Planet, float] = {
        Planet.EARTH: 1,
        Planet.MERCURY: 0.2408467,
        Planet.VENUS: 0.61519726,
        Planet.MARS: 1.8808158,
        Planet.JUPITER: 11.862615,
        Planet.SATURN: 29.447498,
        Planet.URANUS: 84.016846,
        Planet.NEPTUNE: 164.79132
    }

    def __init__(self, seconds: int) -> None:
        self.seconds = seconds

    def on_earth(self) -> float:
        return self.__on(self.Planet.EARTH)

    def on_mercury(self) -> float:
        return self.__on(self.Planet.MERCURY)

    def on_venus(self) -> float:
        return self.__on(self.Planet.VENUS)

    def on_mars(self) -> float:
        return self.__on(self.Planet.MARS)

    def on_jupiter(self) -> float:
        return self.__on(self.Planet.JUPITER)

    def on_saturn(self) -> float:
        return self.__on(self.Planet.SATURN)

    def on_uranus(self) -> float:
        return self.__on(self.Planet.URANUS)

    def on_neptune(self) -> float:
        return self.__on(self.Planet.NEPTUNE)

    def __on(self, planet: Planet) -> float:
        return round(
            self.seconds /
            self.__EARTH_ORBITAL_PERIOD /
            self.__ORBITAL_FACTORS[planet],
            2
        )
